evalml/pipelines/component_graph.py

```diff
@@ -455,8 +465,20 @@ def _transform_features(
                 else:
                     output = component_instance.predict(x_inputs)
                 output_cache[f"{component_name}.x"] = output
+            if self.cached_data is not None and fit:
+                self.component_instances[component_name] = component_instance
+
         return output_cache
 
+    def _get_stacked_ensemble_component(self, hashes, component_name, fit):
```

Comment:
Currently, I didn't want to emit any warnings since it doesn't seem to be useful to the user. If users got warnings, what would they do with the information? Our component graph would continue to train the components as usual if it wasn't able to find them in the cache. However, it could be useful to log it in the debugger (logger.DEBUG?).

As for `fit` being true for cache retrieval, the logic I've put goes as follows:
- If we are fitting, we try to find the appropriately-trained component in the cache. If it exists, we pull that and insert it into our `component_graph.component_instances` dictionary, which we then use for `predict` and other future calls.
- If we are predicting, we expect the fit call to already have been made. Rather than looking at the cache, the properly-fitted components should then already be in the `component_instances` dictionary, and we can operate through that directly.